sdk/typescript/libraries/botbuilder-solutions/src/skills/skillDialog.ts

```diff
@@ -4,334 +4,154 @@
-                    type: ActivityTypes.Trace,
-                    text: `<--Ending the skill conversation with the ${
-                        this.skillManifest.name } Skill and handing off to Parent Bot due to unable to obtain token for user.`
-                });
+            const activity: Activity = <Activity>ActivityEx.createEndOfConversationActivity();
+            this.applyParentActivityProperties(turnContext, activity);
 
-                return await innerDc.endDialog();
-            } else {
-                let dialogResult: DialogTurnResult = {
-                    status: DialogTurnStatus.waiting
-                };
+            await this.sendToSkill(activity, undefined);
+        }
 
-                // if there's any response we need to send to the skill queued
-                // forward to skill and start a new turn
-                while (this.queuedResponses.length > 0) {
-                    const lastEvent: Partial<Activity> | undefined = this.queuedResponses.shift();
+        await super.endDialog(turnContext, instance, reason);
+    }
 
-                    if (lastEvent !== undefined) {
-                        dialogResult = await this.forwardToSkill(innerDc, lastEvent);
-                    }
-                }
+    private applyParentActivityProperties(turnContext: TurnContext, skillActivity: Activity, dialogArgs?: SkillDialogArgs) {
+        // Apply conversation reference and common properties from incoming activity before sending.
+        skillActivity.relatesTo = turnContext.activity.relatesTo;
+        skillActivity.channelData = turnContext.activity.channelData;
+        // skillActivity.properties = turnContext.activity.properties; PENDING
 
-                return dialogResult;
-            }
-        } catch (error) {
-            // something went wrong forwarding to the skill, so end dialog cleanly and throw so the error is logged.
-            // NOTE: errors within the skill itself are handled by the OnTurnError handler on the adapter.
-            await innerDc.endDialog();
-            throw error;
+        if (dialogArgs !== undefined)
+        {
+            skillActivity.value = dialogArgs?.value
         }
     }
 
-    private getTokenRequestCallback(dialogContext: DialogContext): TokenRequestHandler {
-        return async (activity: Activity): Promise<void> => {
-            // Send trace to emulator
-            await dialogContext.context.sendActivity({
-                type: ActivityTypes.Trace,
-                text: '<--Received a Token Request from a skill'
-            });
-
-            const result: DialogTurnResult = await dialogContext.beginDialog(this.authDialog ? this.authDialog.id : '');
-
-            if (result.status === DialogTurnStatus.complete) {
-                const tokenResponse: IProviderTokenResponse = <IProviderTokenResponse> result.result;
+    private async sendToSkill(activity: Activity, dc?: DialogContext): Promise<DialogTurnResult> {
```

Comment:
The parameters are in different order compared to C#, they can be switched, but there's a missconception between optional and nullable, the `dc?: DialogContext` is an optional parameter, but in C# it's treated as nullable. This can be fixed as follows:
```suggestion
    private async sendToSkill(dc: DialogContext | undefined, activity: Activity): Promise<DialogTurnResult> {
```
This would require changing the order of the parameters in all the places that `sendToSkill` is called.

Reply:
Applied the changes

Reply:
The calls to the method still have the old parameter order. When updating a method's firm, you should check all the method's references.